auth-external-api/src/main/java/uk/gov/di/authentication/external/lambda/TokenHandler.java

```diff
@@ -150,12 +158,22 @@ public APIGatewayProxyResponseEvent tokenRequestHandler(APIGatewayProxyRequestEv
 
             authorisationCodeService.updateHasBeenUsed(authCodeStore.getAuthCode(), true);
 
+            String subjectID = authCodeStore.getSubjectID();
+            UserProfile userProfile = dynamoService.getUserProfileFromSubject(subjectID);
+            String internalPairwiseId =
+                    userProfile.getSalt() == null
+                            ? AuditService.UNKNOWN
+                            : ClientSubjectHelper.calculatePairwiseIdentifier(
+                                    subjectID,
+                                    configurationService.getInternalSectorUri(),
+                                    SdkBytes.fromByteBuffer(userProfile.getSalt()).asByteArray());
+
             auditService.submitAuditEvent(
                     AuthExternalApiAuditableEvent.TOKEN_SENT_TO_ORCHESTRATION,
                     AuditService.UNKNOWN,
                     AuditService.UNKNOWN,
                     Optional.ofNullable(requestBody.get("client_id")).orElse(AuditService.UNKNOWN),
-                    Optional.ofNullable(authCodeStore.getSubjectID()).orElse(AuditService.UNKNOWN),
+                    internalPairwiseId,
```

Comment:
Probably best to follow the same pattern to ensure "" is returned rather than "null". So something like `Optional.ofNullable(internalPairwiseId).orElse(AuditService.UNKNOWN)`

Reply:
Thanks mate. Will do that now